### PR TITLE
[SCAL-85659] Support encoding of url params in embed sdk

### PR DESCRIPTION
**Pull request by utsav-test**

### CODE REVIEW

File: src/embed/base.ts [hunk 1]
```diff
@@ -174,6 +174,13 @@ export class TsEmbed {
      */
     private isError: boolean;
 
+    /**
+     * [Basic] Should we encode URL Query Params which thoughtspot will generate for embedding
```
Comment (ashubham) [inline src/embed/base.ts:178]: What does "Basic" mean ? 

Also, we should perhaps have the key `base64Url` either in the the key or in the comments atleast.

File: src/types.ts [hunk 2]
```diff
@@ -90,6 +90,13 @@ export interface EmbedConfig {
 
     /** @internal */
     basepath?: string;
+
+    /**
+     * [Basic] Should we encode URL Query Params which thoughtspot will generate for embedding
```
Comment (ashubham) [inline src/types.ts:95]: This has nothing to do with "Basic" auth.

File: src/embed/base.ts [hunk 3]
```diff
@@ -270,9 +278,14 @@ export class TsEmbed {
...
         const disableProfileAndHelpParam = `&profileAndHelpInNavBarHidden=${disableProfileAndHelp}`;
-        const queryParams = `?embedApp=true${
-            isAppEmbed ? primaryNavParam : ''
-        }${isAppEmbed ? disableProfileAndHelpParam : ''}${queryStringFrag}`;
+        let queryParams = `?embedApp=true${isAppEmbed ? primaryNavParam : ''}${
+            isAppEmbed ? disableProfileAndHelpParam : ''
+        }${queryStringFrag}`;
+        if (this.shouldEncodeUrlQueryParams) {
+            queryParams = `?${getEncodedQueryParamsString(
+                queryParams.substr(1),
```
Comment (ashubham) [inline src/embed/base.ts:286]: Whats the name of the `key` ?

File: src/embed/base.ts (hunk 1 above)
Comment (utsav-test) [reply src/embed/base.ts:178]: Sure will add it to both

File: src/embed/base.ts (hunk 3 above)
Comment (ashubham) [reply src/embed/base.ts:286]: in: embedApp=true&dataSources=["123"]
out: jkshfd234982370lasdsaf

`host?ts64=out#/`

Comment (utsav-test) [reply src/embed/base.ts:287]: Fixed

**Comment by ashubham** [comment]
Also, lets write a unit test for this.

**Review by jbhanu-thoughtspot — APPROVED** [review]

**Pull request merged**